### PR TITLE
Tweaking dev env

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
This is a follow up to #4961, trying to make things easier to setup and lead to less surprises.

* Removes appraisals. We were barely using it, we don't really need it. `BUNDLE_GEMFILE` is all we want, just like Travis.
* Updates Rails 5.1 tests to actually use Rails 5.1. Dependencies were resolving to Rails 5.1.0.rc2...
* Updates other development dependencies.
* Simplify the whole development setup and docs.

### CODE REVIEW

File: gemfiles/rails_51.gemfile [hunk 1]
```diff
@@ -2,10 +2,10 @@ source "https://rubygems.org"
 
 eval_gemfile(File.expand_path(File.join("..", "Gemfile"), __dir__))
 
-gem "rails", "~> 5.1.x"
+gem "rails", "~> 5.1"
 gem "jquery-ui-rails", "~> 5.0"
-gem "devise", "> 4.x"
-gem "draper", "> 3.x"
+gem "devise", git: "https://github.com/plataformatec/devise"
```
Comment (varyonic) [inline gemfiles/rails_51.gemfile]: 4.3 is coming real soon, really https://github.com/plataformatec/devise/issues/4518#issuecomment-300962107

Comment (deivid-rodriguez) [inline gemfiles/rails_51.gemfile]: Cool, we can wait for it! I'm watching that ticket now.

Comment (deivid-rodriguez) [inline gemfiles/rails_51.gemfile]: Updated after devise's release!

**Comment by deivid-rodriguez** [comment]
@Fivell Could you point out which specific changes? I've updated the documentation in that file...

**Comment by Fivell** [comment]
@deivid-rodriguez , since you deleted appraisal 

https://github.com/activeadmin/activeadmin/blob/master/CONTRIBUTING.md#3-get-the-test-suite-running

and

https://github.com/activeadmin/activeadmin/blob/a26ae95af7122e24cca5c129b15c14b55c663a70/CONTRIBUTING.md#7-run-tests-against-major-supported-rails-versions

**Comment by deivid-rodriguez** [comment]
@Fivell please check the diff!

**Comment by Fivell** [comment]
@deivid-rodriguez , I'm blind 😿

**Pull request merged**